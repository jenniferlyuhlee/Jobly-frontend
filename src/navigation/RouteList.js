import React from "react";
import { Routes, Route, Navigate } from "react-router-dom";
import ProtectedRoute from "./ProtectedRoute";
import Home from "../home/Home";
import CompanyList from "../companies/CompaniesList";
import CompanyDetails from "../companies/CompanyDetail";
import JobsList from "../jobs/JobsList";
import ProfileForm from "../auth-user/ProfileForm";
import LoginForm from "../auth-user/LoginForm";
import SignupForm from "../auth-user/SignupForm"

/**
 * Jobly Routes
 * Props: login & signup handling functions inherited from App.js
 * Contains protected routes that check for currUser in state
 */

function RouteList({login, signup}){
    return (
        <div>
            <Routes>
                <Route path="/" element={<Home />}></Route>
                <Route path="/companies" 
                    element={<ProtectedRoute element={<CompanyList />} />} 
                />
                <Route path="/companies/:handle" 
                    element={<ProtectedRoute element={<CompanyDetails />} />}   
                />
                <Route path="/jobs"
                    element={<ProtectedRoute element={<JobsList />} />}   
                />
                <Route path="/profile"
                    element={<ProtectedRoute element={<ProfileForm />} />}   
                />
                <Route path="/signup" element={<SignupForm signup={signup}/>} />
                <Route path="/login" element={<LoginForm login={login}/>} />
                <Route path="*" element={<Navigate to="/"/>} />
            </Routes>
        </div>
    )
}

export default RouteList;