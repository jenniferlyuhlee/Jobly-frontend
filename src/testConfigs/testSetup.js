import React from "react";
import UserContext from "../auth-user/UserContext";

const demoUser = {
    username: "testuser",
    first_name: "testfirst",
    last_name: "testlast",
    email: "test@test.com",
    photo_url: null,
    applications: []
  };

  const UserProvider =
      ({ children, currUser = demoUser, hasAppliedToJob = () => false }) => (
      <UserContext.Provider value={{ currUser, hasAppliedToJob }}>
        {children}
      </UserContext.Provider>
  );

  export {UserProvider};
