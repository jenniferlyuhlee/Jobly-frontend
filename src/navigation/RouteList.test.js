import React from 'react';
import {render} from "@testing-library/react"
import { MemoryRouter } from 'react-router-dom';
import { UserProvider } from '../testConfigs/testSetup';
import RouteList from './RouteList';

it("renders without crashing" , function () {
    render(
        <MemoryRouter>
            <UserProvider>
                <RouteList />
            </UserProvider>
        </MemoryRouter>
    );
});

it("matches snapshot", function () {
    const {asFragment} = render(
        <MemoryRouter>
            <UserProvider>
                <RouteList />
            </UserProvider>
        </MemoryRouter>
    );
    
    expect(asFragment()).toMatchSnapshot();
});