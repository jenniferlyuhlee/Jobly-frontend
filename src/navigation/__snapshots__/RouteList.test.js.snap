// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <div
      class="text-center m-5"
    >
      <div>
        <h1
          class="display-5"
        >
          Welcome back !
        </h1>
        <h5>
          All the jobs in one, covenient place.
        </h5>
        <div
          class="mt-4"
        >
          <p
            class="m-0"
          >
            See 
            <a
              class="tertiary"
              href="/companies"
            >
              who's hiring
            </a>
             â€¢ Explore 
            <a
              class="tertiary"
              href="/jobs"
            >
              new roles
            </a>
          </p>
        </div>
      </div>
      <img
        src="https://png.pngtree.com/png-vector/20220724/ourmid/pngtree-job-choice-icon-job-symbol-apply-vector-png-image_38124340.png"
      />
    </div>
  </div>
</DocumentFragment>
`;
